jboss/container/wildfly/launch/os/node-name/added/launch/openshift-node-name.sh

```diff
@@ -8,9 +12,14 @@ function init_node_name() {
       JBOSS_NODE_NAME="${HOSTNAME}"
     fi
 
-    # CLOUD-427: truncate to 23 characters max (from the end backwards)
+    # CLOUD-2268: truncate to 23 characters max
     if [ ${#JBOSS_NODE_NAME} -gt 23 ]; then
-      JBOSS_NODE_NAME=${JBOSS_NODE_NAME: -23}
+      local prefix=${JBOSS_NODE_NAME::14}
+      local suffix=${JBOSS_NODE_NAME#$prefix}
+      suffix=$(echo -n $suffix | md5sum | cut -c -8 )
+      local jbossNodeNamePrevious="${JBOSS_NODE_NAME}"
+      JBOSS_NODE_NAME="${prefix}-${suffix}"
```

Comment:
@ochaloup If you're looking to reduce collisions, dropping this '-' will give you one more byte for $(echo -n $suffix | md5sum | cut -c -**9** )